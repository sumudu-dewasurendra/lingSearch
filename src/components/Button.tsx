import React from 'react';
import {TouchableOpacity, StyleSheet, Text} from 'react-native';
import {COLORS} from '../theme/colors';

const styles = StyleSheet.create({
  buttonWrapper: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: COLORS.BORDER_COLOR,
    padding: 5,
  },
  buttonText: {
    fontWeight: '700',
    fontSize: 15,
    letterSpacing: 0.5,
  },
});

type ButtonProps = {
  label: string;
  onPressButton: (x?: any) => void;
  width?: number;
  height?: number;
  borderRadius?: number;
  labelColor?: string;
  wrapperStyles?: {[key: string]: any};
};

// Use TouchableOpacity to give the button a touchable effect for the user
const Button = ({
  label,
  onPressButton,
  height,
  width,
  borderRadius,
  wrapperStyles,
  labelColor,
}: ButtonProps) => {
  return (
    <TouchableOpacity
      style={{
        ...styles.buttonWrapper,
        ...wrapperStyles,
        height: height,
        width: width,
        borderRadius: borderRadius,
      }}
      onPress={onPressButton}>
      <Text style={{...styles.buttonText, color: labelColor}}>{label}</Text>
    </TouchableOpacity>
  );
};

export default Button;
