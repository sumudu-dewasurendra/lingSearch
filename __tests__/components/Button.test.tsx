import React from 'react';
import {create} from 'react-test-renderer';
import Button from '../../src/components/Button';
import {expect, test} from '@jest/globals';

test('renders correctly', () => {
  const tree = create(
    <Button
      label="Button"
      onPressButton={() => console.log('Button pressed')}
      width={100}
      height={50}
      borderRadius={5}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
