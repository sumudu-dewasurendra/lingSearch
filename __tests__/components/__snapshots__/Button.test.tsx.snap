// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#c5c5af",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 5,
      "width": 100,
    }
  }
>
  <Text
    style={
      {
        "color": undefined,
        "fontSize": 15,
        "fontWeight": "700",
        "letterSpacing": 0.5,
      }
    }
  >
    Button
  </Text>
</View>
`;
